### PR TITLE
Anomalies can now be triggered at your location with admin_setup. Also sets control on setup() for events

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

Anomaly random events will now ask if you want their anomaly to spawn at your location if triggered via admin setup.

This PR also makes a small but useful change to the logic for new round_event objects. round_events now have their control (their associated round_event_controller) set on New, rather than after setup is called. Previously, round_events could not access control in their setup proc (as it was not set at the time), meaning transferring admin-chosen variables had to be done later in start/announce (depending on which comes sooner).

While this has just been worked around in the past, the change needed to be made here, as some anomaly event types announce first, and some start first. The admin variables needed to be passed in earlier than start/announce (in setup).

Short version: you can access control in round_event setup() now :D
## Why It's Good For The Game

MORE adminbus, MORE anomaly shenanigans (they're one of my favorite event types). Also resolves a quirk in event code that bothered me in other PRs I've made.
## Changelog
:cl: Rhials
admin: anomaly events now give the option to occur at your current location when triggered with the Trigger Event admin verb
/:cl:

### CODE REVIEW

File: code/modules/events/anomaly/_anomaly.dm [hunk 1]
```diff
@@ -7,6 +7,20 @@
 	weight = 15
 	category = EVENT_CATEGORY_ANOMALIES
 	description = "This anomaly shocks and explodes. This is the base type."
+	///The admin-chosen spawn location.
+	var/turf/spawn_location
+	///The area the anomaly will hit in, based on spawn_location
```
Comment (ZephyrTFA) [inline code/modules/events/anomaly/_anomaly.dm:12]: Is the area even needed here since you pass the turf?

**Comment by Jacquerel** [comment]
You _can_ just do this simply by spawning the anomaly from the game panel... but then you'd need to write the announcement yourself so this still seems useful.
